src/components/Profile.vue

```diff
@@ -203,18 +224,65 @@ export default {
         backgroundImage: `url(${avatarUrl})`
       },
       certifications,
-      certKey
+      certKey,
+      errors: [],
+      invalidInputs: [],
+      msg: null
     }
   },
   methods: {
+    /**
+     * Toggle editing state.
+     * {Case A} if activeEdit === false: enter the editing state by setting activeEdit to true
+     * {Case B} if activeEdit === true: save profile changes & exit the editing state by setting activeEdit to false
+     */
     editProfile () {
-      if (this.activeEdit) {
-        UserService.setProfile(this, this.user)
-        this.editBtnMsg = 'Edit Profile'
-        this.activeEdit = false
-      } else {
+      // {Case A} Enter the editing state, then early exit
+      if (!this.activeEdit) {
         this.editBtnMsg = 'Save Profile'
         this.activeEdit = true
+        return
+      }
+
+      // {Case B} The remainder of this function saves new changes and exits the editing state
+
+      // Start by erasing previous errors
+      this.errors = []
+      this.invalidInputs = []
+
+      // Validate fields
+      if (!this.user.isVolunteer && !this.user.highschool) {
+        this.errors.push('Please tell us what high school you go to.')
+        this.invalidInputs.push('highschool')
+      }
+      if (this.user.isVolunteer) {
+        if (!this.user.phone || !phoneValidation.validatePhoneNumber(this.user.phone)) {
+          this.errors.push('Please enter a valid U. S. phone number.')
+          this.invalidInputs.push('phone')
+        }
+        if (!this.user.college) {
+          this.errors.push('Please tell us what college you go to.')
+          this.invalidInputs.push('college')
+        }
+        if (!this.user.favoriteAcademicSubject) {
+          this.errors.push('Please tell us your favorite academic subject.')
+          this.invalidInputs.push('favoriteAcademicSubject')
+        }
+      }
+
+      if (!this.errors.length) {
+        if (!this.msg) {
+          var unwatch = this.$watch('msg', function (newMsg, oldMsg) {
```

Comment:
Could you add some explanatory comments around these blocks of code?

It's not within the scope of this PR, but we should refactor the way this server confirmation stuff is handled. Right now it seems like the `msg` prop is being used to indicate the success of the server request, which is really convoluted, and we should replace all that with promises.

Reply:
I've added a few comments which will hopefully clarify this, and have refactored the UserService.setProfile function so I no longer need to watch the `msg` prop.